packages/eslint-plugin/typings/ts-eslint.d.ts

```diff
@@ -397,8 +397,159 @@ declare module 'ts-eslint' {
+    TSMappedType?: RuleFunction<TSESTree.TSMappedType>;
+    TSMethodSignature?: RuleFunction<TSESTree.TSMethodSignature>;
+    TSModuleBlock?: RuleFunction<TSESTree.TSModuleBlock>;
+    TSModuleDeclaration?: RuleFunction<TSESTree.TSModuleDeclaration>;
+    TSNamespaceExportDeclaration?: RuleFunction<
+      TSESTree.TSNamespaceExportDeclaration
+    >;
+    TSNeverKeyword?: RuleFunction<TSESTree.TSNeverKeyword>;
+    TSNonNullExpression?: RuleFunction<TSESTree.TSNonNullExpression>;
+    TSNullKeyword?: RuleFunction<TSESTree.TSNullKeyword>;
+    TSNumberKeyword?: RuleFunction<TSESTree.TSNumberKeyword>;
+    TSObjectKeyword?: RuleFunction<TSESTree.TSObjectKeyword>;
+    TSOptionalType?: RuleFunction<TSESTree.TSOptionalType>;
+    TSParameterProperty?: RuleFunction<TSESTree.TSParameterProperty>;
+    TSParenthesizedType?: RuleFunction<TSESTree.TSParenthesizedType>;
+    TSPrivateKeyword?: RuleFunction<TSESTree.TSPrivateKeyword>;
+    TSPropertySignature?: RuleFunction<TSESTree.TSPropertySignature>;
+    TSProtectedKeyword?: RuleFunction<TSESTree.TSProtectedKeyword>;
+    TSPublicKeyword?: RuleFunction<TSESTree.TSPublicKeyword>;
+    TSQualifiedName?: RuleFunction<TSESTree.TSQualifiedName>;
+    TSReadonlyKeyword?: RuleFunction<TSESTree.TSReadonlyKeyword>;
+    TSRestType?: RuleFunction<TSESTree.TSRestType>;
+    TSStaticKeyword?: RuleFunction<TSESTree.TSStaticKeyword>;
+    TSStringKeyword?: RuleFunction<TSESTree.TSStringKeyword>;
+    TSSymbolKeyword?: RuleFunction<TSESTree.TSSymbolKeyword>;
+    TSThisType?: RuleFunction<TSESTree.TSThisType>;
+    TSTupleType?: RuleFunction<TSESTree.TSTupleType>;
+    TSTypeAliasDeclaration?: RuleFunction<TSESTree.TSTypeAliasDeclaration>;
+    TSTypeAnnotation?: RuleFunction<TSESTree.TSTypeAnnotation>;
+    TSTypeAssertion?: RuleFunction<TSESTree.TSTypeAssertion>;
+    TSTypeLiteral?: RuleFunction<TSESTree.TSTypeLiteral>;
+    TSTypeOperator?: RuleFunction<TSESTree.TSTypeOperator>;
+    TSTypeParameter?: RuleFunction<TSESTree.TSTypeParameter>;
+    TSTypeParameterDeclaration?: RuleFunction<
+      TSESTree.TSTypeParameterDeclaration
+    >;
+    TSTypeParameterInstantiation?: RuleFunction<
+      TSESTree.TSTypeParameterInstantiation
+    >;
+    TSTypePredicate?: RuleFunction<TSESTree.TSTypePredicate>;
+    TSTypeQuery?: RuleFunction<TSESTree.TSTypeQuery>;
+    TSTypeReference?: RuleFunction<TSESTree.TSTypeReference>;
+    TSUndefinedKeyword?: RuleFunction<TSESTree.TSUndefinedKeyword>;
+    TSUnionType?: RuleFunction<TSESTree.TSUnionType>;
+    TSUnknownKeyword?: RuleFunction<TSESTree.TSUnknownKeyword>;
+    TSVoidKeyword?: RuleFunction<TSESTree.TSVoidKeyword>;
+    VariableDeclaration?: RuleFunction<TSESTree.VariableDeclaration>;
+    VariableDeclarator?: RuleFunction<TSESTree.VariableDeclarator>;
+    WhileStatement?: RuleFunction<TSESTree.WhileStatement>;
+    WithStatement?: RuleFunction<TSESTree.WithStatement>;
+    YieldExpression?: RuleFunction<TSESTree.YieldExpression>;
+  }
```

Comment:
we should add here codePathAnalysis methods
https://eslint.org/docs/developer-guide/code-path-analysis/#events

Reply:
Made extra types for those. Added JSDoc from link you posted.

Reply:
thank you :)